File: config-services/src/main/java/com/microservices/configservices/service/BannerService.java
package com.microservices.configservices.service;

import java.util.List;
import java.util.UUID;
import com.microservices.configservices.payload.request.BannerRequest;
import com.microservices.configservices.payload.response.BannerResponse;


public interface BannerService {

    BannerResponse create(BannerRequest BannerRequest);

    void setImage(UUID id, String image);

    void switchStatus(UUID id);

    void trash(UUID id);

    void isDisplay(UUID id);

    BannerResponse getById(UUID id);

    List<BannerResponse> getAll();

    BannerResponse update(UUID id, BannerRequest BannerRequest);

    BannerResponse delete(UUID id);
}

File: reactjs-app/src/services/BannerService.js
import httpAxios from '../httpAxios';

const BannerService = {

    create: (brandData) => {
        return httpAxios.post(`config-services/api/banners/create`, brandData);
    },
    setImage: (data) => {
        return httpAxios.put(`config-services/api/banners/set-image`, data);
    },
    getById: (id) => {
        return httpAxios.get(`config-services/api/banners/get-by-id/${id}`);
    },
    getAll: () => {
        return httpAxios.get(`config-services/api/banners/get-all`);
    },
    
    update: (id, brand) => { 
        return httpAxios.put(`config-services/api/banners/update/${id}`, brand);
    },
    sitchStatus: (id) => {
        return httpAxios.put(`config-services/api/banners/switch-status/${id}`);
    },
    trash: (id) => {
        return httpAxios.put(`config-services/api/banners/trash/${id}`);
    },
    delete: (id) => {
        return httpAxios.delete(`config-services/api/banners/delete/${id}`);
    },
    display: (id) => {
        return httpAxios.put(`config-services/api/banners/display/${id}`);
    },
};

export default BannerService;
